verl/trainer/fsdp_sft_trainer.py

```diff
@@ -365,7 +365,10 @@ def _compute_loss_and_backward(self, batch, do_backward=True):
 
         # Context manager for sequence parallel if needed
         context = self.sharding_manager if use_sp else nullcontext()
-        with context, torch.autocast(device_type=self.device_name, dtype=torch.bfloat16):
+        # If last micro-batch in an accumulation step, then implicit grad all-reduce happens in loss backward.
+        # Otherwise, use no_sync() to avoid communication overhead during intermediate accumulation steps.
+        sync_context = nullcontext() if last_micro_in_accum else self.fsdp_model.no_sync()
```

Comment:
We don't want to use no_sync here as it will cause higher memory consumption. To prevent silent errors, we disable CPUOffloading explicitly for FSDP1

Reply:
Removed that part of PR